src/api/search_transactions.rs

```diff
@@ -2,75 +2,83 @@ use coinbase_mesh::models::{
   AccountIdentifier, BlockIdentifier, BlockTransaction, SearchTransactionsRequest, SearchTransactionsResponse,
   Transaction, TransactionIdentifier,
 };
+use convert_case::{Case, Casing};
 use serde_json::json;
 use sqlx::FromRow;
 
 use crate::{
-  operation, util::DEFAULT_TOKEN_ID, ChainStatus, MinaMesh, MinaMeshError, OperationType, TransactionStatus,
-  UserCommandType,
+  operation, util::DEFAULT_TOKEN_ID, ChainStatus, InternalCommandType, MinaMesh, MinaMeshError, OperationType,
+  TransactionStatus, UserCommandType,
 };
 
 impl MinaMesh {
   pub async fn search_transactions(
     &self,
     req: SearchTransactionsRequest,
   ) -> Result<SearchTransactionsResponse, MinaMeshError> {
-    let user_commands = self.fetch_user_commands(&req).await?;
-    let user_commands_len = user_commands.len();
-    let next_offset = req.offset.unwrap_or(0) + user_commands_len as i64;
+    let original_offset = req.offset.unwrap_or(0);
+    let mut offset = original_offset;
+    let mut limit = req.limit.unwrap_or(100);
+    let mut transactions = Vec::new() as Vec<BlockTransaction>;
+    let mut txs_len = 0;
+    let mut total_count = 0;
 
-    // Extract the total count from the first user command, or default to 0
+    // User Commands
+    let user_commands = self.fetch_user_commands(&req, offset, limit).await?;
+    let user_commands_len = user_commands.len() as i64;
     let user_commands_total_count = user_commands.first().and_then(|uc| uc.total_count).unwrap_or(0);
+    transactions.extend(user_commands.into_iter().map(|ic| ic.into()));
+    total_count += user_commands_total_count;
+    txs_len += user_commands_len;
 
-    // Map user commands into block transactions
-    let user_commands_bt = user_commands.into_iter().map(|uc| uc.into_block_transaction()).collect();
+    // Internal Commands
+    if limit > total_count {
+      // if we are below the limit, fetch internal commands
+      (offset, limit) = adjust_limit_and_offset(limit, offset, txs_len);
+      let internal_commands = self.fetch_internal_commands(&req, offset, limit).await?;
+      let internal_commands_len = internal_commands.len() as i64;
+      let internal_commands_total_count = internal_commands.first().and_then(|ic| ic.total_count).unwrap_or(0);
+      transactions.extend(internal_commands.into_iter().map(|uc| uc.into()));
+      txs_len += internal_commands_len;
+      total_count += internal_commands_total_count;
```

Comment:
Could/should this be extracted into its own fn and reused above (and possibly below as well)?